src/control/cmd/daos/filesystem.go

```diff
@@ -265,3 +269,151 @@ func (cmd *fsGetAttrCmd) Execute(_ []string) error {
 
 	return nil
 }
+
+type fsCheckCmd struct {
+	existingContainerCmd
+
+	FsckFlags FsCheckFlag `long:"flags" short:"f" description:"comma-separated flags: print, remove, relink, or verify"`
+	DirName   string      `long:"dir-name" short:"n" description:"directory name under lost+found to store leaked oids (a timestamp dir would be created if this is not specified)"`
+}
+
+func (cmd *fsCheckCmd) Execute(_ []string) error {
+	ap, deallocCmdArgs, err := allocCmdArgs(cmd.Logger)
+	if err != nil {
+		return err
+	}
+	defer deallocCmdArgs()
+
+	if err := cmd.resolveContainer(ap); err != nil {
+		return err
+	}
+
+	var cleanupPool func()
+	cleanupPool, err = cmd.connectPool(C.DAOS_PC_RW, ap)
+	if err != nil {
+		return err
+	}
+	defer cleanupPool()
+
+	rc := C.dfs_cont_check(cmd.cPoolHandle, &ap.cont_str[0], cmd.FsckFlags.Flags,
+		C.CString(cmd.DirName))
+	if err := dfsError(rc); err != nil {
+		return errors.Wrapf(err, "failed filesystem check")
+	}
+
+	return nil
+}
+
+type fsFixChunkSizeCmd struct {
+	fsAttrCmd
+
+	ChunkSize ChunkSizeFlag `long:"chunk-size" short:"z" description:"actual file chunk size" required:"1"`
+}
+
+func (cmd *fsFixChunkSizeCmd) Execute(_ []string) error {
+	ap, deallocCmdArgs, err := setupFSAttrCmd(&cmd.fsAttrCmd)
+	if err != nil {
+		return err
+	}
+	defer deallocCmdArgs()
+
+	flags := C.uint(C.DAOS_COO_RW)
+
+	ap.fs_op = C.FS_CHECK
+	cleanup, err := cmd.resolveAndConnect(flags, ap)
+	if err != nil {
+		return err
+	}
+	defer cleanup()
+
+	if cmd.ChunkSize.Set {
+		ap.chunk_size = cmd.ChunkSize.Size
+	}
+
+	if err := dfsError(C.fs_set_cs_hdlr(ap)); err != nil {
+		return errors.Wrapf(err, "Fix Chunk size failed")
+	}
+
+	return nil
+}
+
+type fsFixSB struct {
+	existingContainerCmd
+
+	Mode            ConsModeFlag  `long:"mode" short:"M" description:"DFS consistency mode"`
+	ChunkSize       ChunkSizeFlag `long:"chunk-size" short:"z" description:"container chunk size"`
+	ObjectClass     ObjClassFlag  `long:"oclass" short:"o" description:"default object class"`
+	DirObjectClass  ObjClassFlag  `long:"dir-oclass" short:"D" description:"default directory object class"`
+	FileObjectClass ObjClassFlag  `long:"file-oclass" short:"F" description:"default file object class"`
+	CHints          string        `long:"hints" short:"h" description:"container hints"`
```

Comment:
This `-h` will conflict with the top-level `--help, -h` flag. Maybe `--hints, -H` instead?

Reply:
ah bummer.. we added this in master for the container create flags and it has -h, so probably should fix that there too